doc/tools/docgen_cmd.py

```diff
@@ -108,43 +106,62 @@ def get_rst_string(module_name, help_string):
     # generate docs
     command_list = []
 
-    workflows_folder = pjoin('..', 'dipy', 'workflows')
-    workflow_class = module = importlib.import_module(
-        "dipy.workflows.workflow")
-
-    for f in listdir(workflows_folder):
-        fpath = pjoin(workflows_folder, f)
-        module_name = inspect.getmodulename(fpath)
-        if module_name is not None:
-            module = importlib.import_module(
-                "dipy.workflows." + module_name)
-            members = inspect.getmembers(module)
-            for member_name, member_obj in members:
-                if(inspect.isclass(member_obj)):
-                    if (issubclass(member_obj, workflow_class.Workflow) and
-                            not member_obj == workflow_class.Workflow):
-                        # member_obj is a workflow
-                        print("Generating docs for: ", member_name)
-                        if hasattr(member_obj, 'run'):
-                            help_string = inspect.getdoc(member_obj.run)
-
-                            doc_string = get_rst_string(member_name,
-                                                        help_string)
-                            out_f = member_name + ".rst"
-                            output_file = open(pjoin(outdir, out_f), "w")
-                            output_file.write(doc_string)
-                            output_file.close()
-                            command_list.append(out_f)
-                            print("Done")
+    workflows_folder = pjoin('..', 'bin')
+    workflow_module = importlib.import_module("dipy.workflows.workflow")
+
+    workflow_flist = [os.path.abspath(pjoin(workflows_folder, f))
+                      for f in os.listdir(workflows_folder)
+                      if os.path.isfile(pjoin(workflows_folder, f)) and
+                      f.lower().startswith("dipy_")]
+
+    workflow_desc = {}
+    # We get all workflows class obj in a dictionnary
+    for f in os.listdir(pjoin('..', 'dipy', 'workflows')):
+        module_name = inspect.getmodulename(f)
+        if module_name is None:
+            continue
+
+        module = importlib.import_module("dipy.workflows." + module_name)
+        members = inspect.getmembers(module)
+        d_wkflw = {name: {"module": obj, "help": get_help_string(obj)}
+                   for name, obj in members
+                   if inspect.isclass(obj) and
+                   issubclass(obj, workflow_module.Workflow) and
+                   name not in ['Workflow', 'CombinedWorkflow']
```

Comment:
Maybe put this list somewhere more prominent as a module-level constant? In case we need to find it and add more items in the future?

Reply:
Good idea, will do, Thanks!